x/username/handlers.go

```diff
@@ -0,0 +1,161 @@
+package username
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+)
+
+const (
+	registerTokenCost     = 0
+	changeTokenOwnerCost  = 0
+	changeTokenTargetCost = 0
+)
+
+func RegisterRoutes(r weave.Registry, auth x.Authenticator) {
+	r = migration.SchemaMigratingRegistry("username", r)
+
+	b := NewTokenBucket()
+	r.Handle(RegisterTokenMsg{}.Path(), &registerTokenHandler{auth: auth, bucket: b})
+	r.Handle(ChangeTokenOwnerMsg{}.Path(), &changeTokenOwnerHandler{auth: auth, bucket: b})
+	r.Handle(ChangeTokenTargetMsg{}.Path(), &changeTokenTargetHandler{auth: auth, bucket: b})
+}
+
+type registerTokenHandler struct {
+	auth   x.Authenticator
+	bucket orm.ModelBucket
+}
+
+func (h *registerTokenHandler) Check(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*weave.CheckResult, error) {
```

Comment:
Very nice, clean, simple implementations of all these handlers. After this weekend, I was really looking for shirt contracts. This is nice example in any case. Even if we don't use this in Bnsd, it is a wonderful example for reference or tutorial